### PR TITLE
[Job Launcher] Added logic for rate processing

### PR DESCRIPTION
**Pull request by eugenvoronov**
## Description
When depositing funds to the balance, the currency deposit rate to USD is taken into account.

## Summary of changes
- Added account of the rate when replenishing the balance
- Added check for a token symbol when replenishing with crypto
- Added `payment` constants
- Added `payment` interface
- Updated tests

## Related issues
[[Job Launcher] - [Payments] - Correctly handle conversion to HMT using rate#650](https://github.com/humanprotocol/human-protocol/issues/650)

### CODE REVIEW

File: packages/apps/job-launcher/server/src/modules/payment/payment.dto.ts [hunk 1]
```diff
@@ -43,18 +42,16 @@ export class PaymentCryptoCreateDto {
 export class PaymentCreateDto {
   public transactionHash?: string;
   public amount?: string;
-  public currency?: Currency;
+  public currency?: string;
```
Comment (flopez7) [inline packages/apps/job-launcher/server/src/modules/payment/payment.dto.ts:45]: Why did you remove the Currency type?

File: packages/apps/job-launcher/server/src/modules/payment/payment.service.ts [hunk 2]
```diff
@@ -164,14 +193,23 @@ export class PaymentService {
...
+      TokenId.HMT,
       PaymentType.DEPOSIT,
-      BigNumber.from(amount),
+      amount,
+      rate,
+      transaction.transactionHash
     );
 
     return true;
+    return true;
```
Comment (flopez7) [inline packages/apps/job-launcher/server/src/modules/payment/payment.service.ts:212]: duplicated return

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts [hunk 3]
```diff
@@ -146,11 +149,18 @@ export class JobService {
       throw new NotFoundException(ErrorJob.NotCreated);
     }
 
+    const rate = await this.currencyService.getRate(
```
Comment (portuu3) [inline packages/apps/job-launcher/server/src/modules/job/job.service.ts:152]: getRate is always called to be passed to savePayment, I think we can move getRate call insice saveParyment function

**Review by portuu3 — CHANGES_REQUESTED** [review]
Check also @flopez7 comments

File: packages/apps/job-launcher/server/src/modules/payment/payment.dto.ts (hunk 1 above)
Comment (eugenvoronov) [reply packages/apps/job-launcher/server/src/modules/payment/payment.dto.ts:45]: It was incorrect a bit, we can store payment in fiat and in crypto, that means currency represent as fiat currencies and tokens symbols at the same time. I don't actually want to use double types for property, for example `currency: Currency | TokenId`, at the same time not good idea to union Currency and TokenId to the same enum type. Probably it can be improved somehow.

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts (hunk 3 above)
Comment (eugenvoronov) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:152]: Good idea, but some times we are using reverse rate as here. What do you think about this case?

File: packages/apps/job-launcher/server/src/modules/payment/payment.service.ts [hunk 4]
```diff
@@ -105,12 +109,19 @@ export class PaymentService {
       throw new BadRequestException(ErrorPayment.NotSuccess);
     }
 
-    await this.savePayment(
-      userId,
-      PaymentSource.FIAT,
-      PaymentType.DEPOSIT,
-      BigNumber.from(paymentData.amount),
-    );
+      const rate = 1 / await this.currencyService.getRate(
```
Comment (eugenvoronov) [inline packages/apps/job-launcher/server/src/modules/payment/payment.service.ts:112]: Reverse rate

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts (hunk 3 above)
Comment (eugenvoronov) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:152]: https://github.com/humanprotocol/human-protocol/pull/654/files#r1263866867

Comment (portuu3) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:152]: would swap the params `from` and `to` work?

Comment (eugenvoronov) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:152]: > would swap the params from and to work?

Comment (eugenvoronov) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:152]: Actually not, because of coingecko api, it will return correct response when token id it's first parameter. For example: 
✅ https://api.coingecko.com/api/v3/simple/price?ids=human-protocol&vs_currencies=usd
❌ https://api.coingecko.com/api/v3/simple/price?ids=usd&vs_currencies=human-protocol

@portuu3

Comment (portuu3) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:152]: Can we have an enumerable of currencies and another one of tokens, so when the getRate function is called, depending on whether 'from' is a currency or a token, we reverse it, would that make sense?

Comment (eugenvoronov) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:152]: Yes, I will implement it

**Review by portuu3 — CHANGES_REQUESTED** [review]

**Review by portuu3 — APPROVED** [review]
